liquibase-extension-testing/src/main/java/liquibase/extension/testing/testsystem/spock/LiquibaseIntegrationMethodInterceptor.java

```diff
@@ -0,0 +1,70 @@
+package liquibase.extension.testing.testsystem.spock;
+
+import liquibase.extension.testing.testsystem.TestSystem;
+import org.spockframework.runtime.extension.AbstractMethodInterceptor;
+import org.spockframework.runtime.extension.IMethodInvocation;
+import org.spockframework.runtime.model.FieldInfo;
+import org.spockframework.runtime.model.SpecInfo;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class LiquibaseIntegrationMethodInterceptor extends AbstractMethodInterceptor {
+
+    private final SpecInfo spec;
+    private final LiquibaseIntegrationTestExtension.ErrorListener errorListener;
+
+    LiquibaseIntegrationMethodInterceptor(SpecInfo spec, LiquibaseIntegrationTestExtension.ErrorListener errorListener) {
+        this.spec = spec;
+        this.errorListener = errorListener;
+    }
+
+    @Override
+    public void interceptSetupSpecMethod(IMethodInvocation invocation) throws Throwable {
+        final List<FieldInfo> containers = findAllContainers();
+        startContainers(containers, invocation);
+
+        invocation.proceed();
+    }
+
+    @Override
+    public void interceptCleanupSpecMethod(IMethodInvocation invocation) throws Throwable {
+        final List<FieldInfo> containers = findAllContainers();
+        stopContainers(containers, invocation);
+
+        invocation.proceed();
+    }
+
+
+    private List<FieldInfo> findAllContainers() {
+        List<FieldInfo> returnList = new ArrayList<>();
+        for (FieldInfo fieldInfo : spec.getAllFields()) {
+            if (TestSystem.class.isAssignableFrom(fieldInfo.getType())) {
+                assert fieldInfo.isShared() : "TestEnvironment field " + fieldInfo.getName() + " must be @Shared";
+                returnList.add(fieldInfo);
+            }
+        }
+        return returnList;
+    }
+
+    private static void startContainers(List<FieldInfo> containers, IMethodInvocation invocation) throws Exception {
+        for (FieldInfo field : containers) {
+            TestSystem env = readContainerFromField(field, invocation);
+            env.start();
+
+        }
+    }
+
+    private void stopContainers(List<FieldInfo> containers, IMethodInvocation invocation) throws Exception {
+        for (FieldInfo field : containers) {
+            TestSystem env = readContainerFromField(field, invocation);
+
+            env.stop();
```

Comment:
Shouldn't you wrap line 62 in a try/catch? If there are multiple containers to stop, and the first one fails, the other ones will not stop.

The same problem will occur with the startContainers method, but it seems like a desirable side effect there.

Reply:
Yeah, I added try catch to just 62 and not the start(). 

This whole spock interceptor is something I got going enough to see how it works but haven't gotten a chance to really use and vet yet. There are likely other improvements we'll need to use once we start wrting tests with it. The existing integration tests were junit